pcsx2/ImGui/ImGuiOverlays.h

```diff
@@ -27,3 +27,16 @@ namespace SaveStateSelectorUI
 	void LoadCurrentSlot();
 	void SaveCurrentSlot();
 } // namespace SaveStateSelectorUI
+
+namespace InputRecordingUI
+{
+	struct InputRecordingData
```

Comment:
Simply having a function that passes strings across would make more sense than exposing private details in a structure.

Reply:
> Simply having a function that passes strings across would make more sense than exposing private details in a structure.

I'm really not sure how to go about this. Ideally what I would want do to make things perfectly atomic is to have the GS, when it needs to print those strings, stop and request them from the EE, waiting until it can provide them in full. I believe right now my code technically has the issue that the GS thread can be trying to read from the struct while the EE thread is trying to write to it. We of course only have one writer, but I still think we nonetheless run into the readers-writers problem. Basically, what I would like to do ideally is:

* GS notices it needs these three strings.
* GS sends a request off to the EE to give it these three strings.
* GS blocks while it waits for the EE to get to a point where it can compose the strings.
* EE composes and sends the strings to the GS, resuming its own work.
* GS unblocks and prints the strings to the ImGui overlay, continuing on from there.

Does this seem feasible?

Reply:
Just replace the struct with a function that takes the 4 arguments (e.g. `InputRecordingUI::SetRecordingData()), then store it to static variables in the function body.

Since you're using `RunOnGSThread()`, the update will be atomic.